cli/boards.go

```diff
@@ -0,0 +1,234 @@
+	// so we are sending in one chunk.
+	// If the file is too big, return error.
+	if file.Len() > boardUploadChunkSize {
+		return nil, fmt.Errorf("file is too large, must be under %d bytes", boardUploadChunkSize)
+	}
+
+	stream, err := c.packageClient.CreatePackage(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error starting CreatePackage stream")
+	}
+
+	stats, err := jsonFile.Stat()
+	if err != nil {
+		return nil, err
+	}
+	boardDefsFile := []*packagepb.FileInfo{{Name: name, Size: uint64(stats.Size())}}
+
+	packageInfo := &packagepb.PackageInfo{
+		OrganizationId: orgID,
+		Name:           name,
+		Version:        version,
+		Type:           packagepb.PackageType_PACKAGE_TYPE_BOARD_DEFS,
+		Files:          boardDefsFile,
+		Metadata:       nil,
+	}
+
+	// send the package requests
+	var errs error
+	if err := sendPackageRequests(stream, file, packageInfo); err != nil {
+		errs = multierr.Combine(errs, errors.Wrapf(err, "error syncing package"))
+	}
+
+	// close the stream and receive a response when finished.
+	resp, err := stream.CloseAndRecv()
+	if err != nil {
+		errs = multierr.Combine(errs, errors.Wrapf(err, "received error response while syncing package"))
+	}
+	if errs != nil {
+		return nil, errs
+	}
+
+	return resp, nil
+}
+
+// helper function to check if a package with this name and version already exists.
+func (c *appClient) boardDefsPackageExists(ctx context.Context, orgID, name, version string) error {
+	// the packageID is the orgid/name
+	packageID := fmt.Sprintf("%s/%s", orgID, name)
+
+	req := packagepb.GetPackageRequest{
+		Id:      packageID,
+		Version: version,
+	}
+
+	_, err := c.packageClient.GetPackage(ctx, &req)
+
+	if err == nil {
+		return fmt.Errorf("a package with name %s and version %s already exists", name, version)
+	}
+	return nil
+}
+
+func sendPackageRequests(stream packagepb.PackageService_CreatePackageClient,
+	f *bytes.Buffer, packageInfo *packagepb.PackageInfo,
+) error {
+	req := &packagepb.CreatePackageRequest{
+		Package: &packagepb.CreatePackageRequest_Info{Info: packageInfo},
+	}
+	// first send the packageinfo.
+	if err := stream.Send(req); err != nil {
+		return err
+	}
+
+	//nolint:errcheck
+	defer stream.CloseSend()
+
+	// Now send the file contents.
+	req = &packagepb.CreatePackageRequest{
+		Package: &packagepb.CreatePackageRequest_Contents{Contents: f.Bytes()},
+	}
+
+	if err := stream.Send(req); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateArchive creates a tar.gz from the file provided.
+func CreateArchive(file *os.File) (*bytes.Buffer, error) {
+	// Create output buffer
+	out := new(bytes.Buffer)
+
+	// These writers are chained. Writing to the tar writer will
+	// write to the gzip writer which in turn will write to
+	// the "out" writer
+	gw := gzip.NewWriter(out)
+	//nolint:errcheck
+	defer gw.Close()
+	tw := tar.NewWriter(gw)
+	//nolint:errcheck
+	defer tw.Close()
```

Comment:
Same suggestions on both of these to get rid of the `nolint` directive and use `utils.UncheckedErrorFunc`.